tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala

```diff
@@ -40,4 +47,452 @@ trait LinagoraCalendarEventParseMethodContract {
+  }
+
+  @Test
+  def parseShouldReturnUnknownMethodWhenMissingOneCapability(): Unit = {
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core"],
+         |  "methodCalls": [[
+         |    "CalendarEvent/parse",
+         |    {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "blobIds": [ "123" ]
+         |    },
+         |    "c1"]]
+         |}""".stripMargin
+
+    val response = `given`
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |    "error",
+           |    {
+           |      "type": "unknownMethod",
+           |      "description": "Missing capability(ies): com:linagora:params:calendar:event"
+           |    },
+           |    "c1"]""".stripMargin)
+  }
+
+  @Test
+  def parseShouldReturnUnknownMethodWhenMissingAllCapabilities(): Unit = {
+    val request: String =
+      s"""{
+         |  "using": [],
+         |  "methodCalls": [[
+         |    "CalendarEvent/parse",
+         |    {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "blobIds": [ "123" ]
+         |    },
+         |    "c1"]]
+         |}""".stripMargin
+
+    val response = `given`
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |    "error",
+           |    {
+           |      "type": "unknownMethod",
+           |      "description": "Missing capability(ies): urn:ietf:params:jmap:core, com:linagora:params:calendar:event"
+           |    },
+           |    "c1"]""".stripMargin)
+  }
+
+  @Test
+  def parseShouldFailWhenWrongAccountId(): Unit = {
+    val request: String =
+      s"""{
+         |  "using": [
+         |    "urn:ietf:params:jmap:core",
+         |    "com:linagora:params:calendar:event"],
+         |  "methodCalls": [[
+         |    "CalendarEvent/parse",
+         |    {
+         |      "accountId": "unknownAccountId",
+         |      "blobIds": [ "0f9f65ab-dc7b-4146-850f-6e4881093965" ]
+         |    },
+         |    "c1"]]
+         |}""".stripMargin
+
+    val response = `given`
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |    "error",
+           |    {
+           |        "type": "accountNotFound"
+           |    },
+           |    "c1"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def parseShouldNotParseBlobEntryWhenDoesNotPermission(server: GuiceJamesServer): Unit = {
```

Comment:
```suggestion
  def parseShouldNotParseBlobEntryWhenDoesNotHavePermission(server: GuiceJamesServer): Unit = {

Reply:
Should we have a test for delegation?

Reply:
For shared mailbox also?

Reply:
the ICS blob is not reference to the mailbox,
I don't find a way how to test it

Reply:
``` 
GIVEN bob shares his mailbox abc to Alice
AND email1 in ABC have an ICS attachment
THEN Alice should the able to EventCalendar/parse this attachment
```